lib/std/rand/Romu192.zig

```diff
@@ -0,0 +1,134 @@
+// *** HEADER COPYED FROM REF. IMPLEMENTATION ***
+// Romu Pseudorandom Number Generators
+//
+// Copyright 2020 Mark A. Overton
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
```

Comment:
Zig uses the MIT license; having files with a different license is probably not desirable.

Reply:
that is not the license of my code, it is simple an header in the paper author code.

Reply:
Best option is to ask author to change license to MIT, since it includes to credit author and it does not require additional maintenance of "how work was modified" etc.
It is disappointing, that author was not able to republish the updates with license change to arxiv and neither completed or responded negatively to the statistical proof as requested on hackernews.

Reply:
Apparently the paper author (PA) has admitted that his RNG is not amenable to mathematical analysis.
Then the PA, to compensate for the lack of a mathematical proof, has done a full enumeration study on a lower size generator (32 bits state), then provided an empirical upper bounds for failure probability on the lower size generator and by a scaling argument extrapolated the results to the larger size generators. Partially addressing C.L. criticisms from a computational stand-point. Generally I agree that his analysis of the RNG is lacking, I only trust the fact that this generator passes PractRand and BigCrush.

Reply:
In any case, since my code has been written from the ground up in Zig I removed the copied license header and only left mention of the paper and the website

Reply:
If you are reimplementing the same algorithm, this code is a derivative, and therefore under the same license. You can't just remove the copyright notice and license like that.

Reply:
Yep correct. The fact that @francescoalemanno copied the license header into their code indicates to me that they did, in fact, refer to the reference implementation while implementing this. MIT-licensed, MIT-compatible or public-domain code would be better suited to inclusion. If we believe there is good reason to include this project in the Zig std library, then perhaps it would be worth contacting the author for permission, or for them to relicense into the public domain or something MIT-compatible.

Reply:
to be honest guys there was no way in hell for me not to look at code... the paper describes the algorithm via code...

Reply:
That doesn't change anything about licensing. You should revert 34595e7 and contact the author about licensing if you want this merged.

Reply:
:crossed_fingers:

Reply:
@francescoalemanno please ignore any suggestions about licenses that do not come from the [core zig team members](https://github.com/orgs/ziglang/people). The license header should be removed because the zig code that you wrote is copyrighted by yourself, not by someone else. Mark A. Overton does not own the copyright to the zig code that you created here. I suggest to simply state in the file something like this:

```zig
// Implemented based on this paper <link>
// Thank you to authors of the paper <name>, <name>, etc.
```

Reply:
i am not a lawyer, but surely a re-implementation of an algorithm, based off an existing implementation, is a derivative of said implementation, and therefore must follow the apache-2 license terms?

Reply:
No that is incorrect.